phoenix-core/src/main/java/org/apache/phoenix/schema/json/PhoenixJson.java

```diff
@@ -0,0 +1,280 @@
+        ObjectMapper objectMapper = new ObjectMapper();
+        try {
+            JsonNode rootNode = objectMapper.readTree(jsonParser);
+            PhoenixJson phoenixJson = new PhoenixJson(rootNode);
+            return phoenixJson;
+        } finally {
+            jsonParser.close();
+        }
+    }
+
+    /* Default for unit testing */PhoenixJson(final JsonNode node) {
+        Preconditions.checkNotNull(node, "root node cannot be null for json");
+        this.node = node;
+    }
+
+    /**
+     * Get {@link PhoenixJson} for a given json paths. For example :
+     * <p>
+     * <code>
+     * {"f2":{"f3":1},"f4":{"f5":99,"f6":{"f7":"2"}}}'
+     * </code>
+     * <p>
+     * for this source json, if we want to know the json at path {'f4','f6'} it will return
+     * {@link PhoenixJson} object for json {"f7":"2"}. It always returns the last key if same key
+     * exist more than once.
+     * <p>
+     * If the given path is unreachable then it throws {@link PhoenixJsonException} with the message
+     * having information about not found path. It is caller responsibility to wrap it in
+     * {@link SQLException} or catch it and return null to client.
+     * @param paths {@link String []} of path in the same order as they appear in json.
+     * @return {@link PhoenixJson} for the json against @paths.
+     * @throws PhoenixJsonException
+     */
+    public PhoenixJson getPhoenixJson(String[] paths) throws PhoenixJsonException {
+        JsonNode node = this.node;
+        for (String path : paths) {
+            JsonNode nodeTemp = null;
+            if (node.isArray()) {
+                try {
+                    int index = Integer.parseInt(path);
+                    nodeTemp = node.path(index);
+                } catch (NumberFormatException nfe) {
+                    throw new PhoenixJsonException("path: " + path + " not found", nfe);
+                }
+            } else {
+                nodeTemp = node.path(path);
+            }
+            if (nodeTemp == null || nodeTemp.isMissingNode()) {
+                throw new PhoenixJsonException("path: " + path + " not found");
+            }
+            node = nodeTemp;
+        }
+        return new PhoenixJson(node);
+    }
+
+    /**
+     * Get {@link PhoenixJson} for a given json paths. For example :
+     * <p>
+     * <code>
+     * {"f2":{"f3":1},"f4":{"f5":99,"f6":{"f7":"2"}}}'
+     * </code>
+     * <p>
+     * for this source json, if we want to know the json at path {'f4','f6'} it will return
+     * {@link PhoenixJson} object for json {"f7":"2"}. It always returns the last key if same key
+     * exist more than once.
+     * <p>
+     * If the given path is unreachable then it return null.
+     * @param paths {@link String []} of path in the same order as they appear in json.
+     * @return {@link PhoenixJson} for the json against @paths.
+     */
+    public PhoenixJson getNullablePhoenixJson(String[] paths) {
```

Comment:
Since getPhoenixJson and getNullablePhoenixJson are almost the same, maybe have a helper function that takes a nullable argument in order to prevent code duplication.